pkg/reconciler/contour/controller.go

```diff
@@ -71,11 +72,24 @@ func NewController(
 		recorder: record.NewBroadcaster().NewRecorder(
 			scheme.Scheme, corev1.EventSource{Component: controllerAgentName}),
 	}
-	impl := controller.NewImpl(c, logger, "ContourIngresses")
+	myFilterFunc := reconciler.AnnotationFilterFunc(networking.IngressClassAnnotationKey, ContourIngressClassName, false)
+	impl := ingressreconciler.NewImpl(ctx, c, func(impl *controller.Impl) controller.Options {
```

Comment:
I often run this SxS with Istio, and we'll need the class-based annotation support for that to work properly, I think.

Reply:
Yes in reconciler contour ingress still filters the ingress objects with their respective class annotations.

Reply:
Well, my point is that now that Istio adds finalizers everywhere we'll get them even if we don't use Istio.  We should update this now that your change has landed 🎉